### PR TITLE
Closes #1793

### LINKED ISSUE #1793: [DEFECT] import (discovery) of RAVEN ``contrib`` seems to be compromised
### Thank you for the defect report

- [X] I am using the latest version of `RAVEN`.
- [X] I have read the [Wiki](https://github.com/idaholab/raven/wiki).
- [X] I have created a [minimum, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)
that demonstrates the defect.

### Defect Description

I did a fresh installation of RAVEN and the execution of tests cases result in an ```import``` error:
```
 ... successfully imported "ExamplePlugin" ...
Traceback (most recent call last):
  File "/Users/alfoa/projects/raven/framework/Driver.py", line 31, in <module>
    from Simulation import Simulation
  File "/Users/alfoa/projects/raven/framework/Simulation.py", line 28, in <module>
    import Steps
  File "/Users/alfoa/projects/raven/framework/Steps/__init__.py", line 22, in <module>
    from .Factory import factory
  File "/Users/alfoa/projects/raven/framework/Steps/Factory.py", line 24, in <module>
    from .SingleRun import SingleRun
  File "/Users/alfoa/projects/raven/framework/Steps/SingleRun.py", line 30, in <module>
    import Models
  File "/Users/alfoa/projects/raven/framework/Models/__init__.py", line 25, in <module>
    from .Dummy import Dummy
  File "/Users/alfoa/projects/raven/framework/Models/Dummy.py", line 31, in <module>
    from Decorators.Parallelization import Parallel
  File "/Users/alfoa/projects/raven/framework/Decorators/Parallelization.py", line 20, in <module>
    from utils import importerUtils as im
  File "/Users/alfoa/projects/raven/framework/utils/importerUtils.py", line 26, in <module>
    import contrib.lazy.lazy_loader as lazy_loader
ModuleNotFoundError: No module named 'contrib.lazy'
```

This issue seems to be triggered by the modifications added in https://github.com/idaholab/raven/pull/1685 (even if it does not seem clear why to me)

### Steps to Reproduce

Fresh installation of RAVEN.

### Expected Behavior

Correctly import RAVEN dependencies in ```contrib```

### Screenshots and Input Files

```
 ... successfully imported "ExamplePlugin" ...
Traceback (most recent call last):
  File "/Users/alfoa/projects/raven/framework/Driver.py", line 31, in <module>
    from Simulation import Simulation
  File "/Users/alfoa/projects/raven/framework/Simulation.py", line 28, in <module>
    import Steps
  File "/Users/alfoa/projects/raven/framework/Steps/__init__.py", line 22, in <module>
    from .Factory import factory
  File "/Users/alfoa/projects/raven/framework/Steps/Factory.py", line 24, in <module>
    from .SingleRun import SingleRun
  File "/Users/alfoa/projects/raven/framework/Steps/SingleRun.py", line 30, in <module>
    import Models
  File "/Users/alfoa/projects/raven/framework/Models/__init__.py", line 25, in <module>
    from .Dummy import Dummy
  File "/Users/alfoa/projects/raven/framework/Models/Dummy.py", line 31, in <module>
    from Decorators.Parallelization import Parallel
  File "/Users/alfoa/projects/raven/framework/Decorators/Parallelization.py", line 20, in <module>
    from utils import importerUtils as im
  File "/Users/alfoa/projects/raven/framework/utils/importerUtils.py", line 26, in <module>
    import contrib.lazy.lazy_loader as lazy_loader
ModuleNotFoundError: No module named 'contrib.lazy'
```

### OS

MacOS

### OS Version

_No response_

### Dependency Manager

CONDA

### For Change Control Board: Issue Review

- [X] Is it tagged with a type: defect or task?
- [X] Is it tagged with a priority: critical, normal or minor?
- [X] If it will impact requirements or requirements tests, is it tagged with requirements?
- [X] If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [X] Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

### For Change Control Board: Issue Closure

- [ ] If the issue is a defect, is the defect fixed?
- [ ] If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [ ] If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by aalfonsi**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #1793

##### What are the significant changes in functionality due to this change request?
added ```__init__``` file in contrib directory

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [ ] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: framework/contrib/pyDOE/build_regression_matrix.py [hunk 1]
```diff
@@ -16,6 +16,14 @@
...
 
 def grep(haystack, needle):
+    """
+    Grep element from haystack
+    
+    Returns
+    -------
+    start : int
+    
+    """
```
Comment (wangcj05) [inline framework/contrib/pyDOE/build_regression_matrix.py:26]: Did you update the doc by yourself? If so, could you add more details about the input parameters?

**Review by wangcj05 — COMMENTED** [review]
One minor comment for your consideration. This PR contains some cleanup and __init__ method update. @aalfonsi

File: framework/contrib/pyDOE/build_regression_matrix.py (hunk 1 above)
Comment (aalfonsi) [reply framework/contrib/pyDOE/build_regression_matrix.py]: yes...pylint was complaining about this docstrings (I am confused on why this has not been captured before from the pyling...did the "testing" settings change?

Comment (aalfonsi) [reply framework/contrib/pyDOE/build_regression_matrix.py]: Addressed

**Comment by aalfonsi** [comment]
@wangcj05 done

**Comment by aalfonsi** [comment]
I ll try today

**Comment by aalfonsi** [comment]
New installation works.Closing.

**Pull request closed without merging**